core/src/main/java/org/mobilitydata/gtfsvalidator/input/CountryCode.java

```diff
@@ -0,0 +1,103 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.input;
+
+import com.google.common.collect.ImmutableSet;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.Set;
+
+/** Represents the country code of a GTFS feed. */
+public class CountryCode {
+  private static final Set<String> ISO_COUNTRIES = ImmutableSet.copyOf(Locale.getISOCountries());
+  private static final String INVALID_COUNTRY_CODE_MESSAGE =
+      "Country code must follow format `cc`, where cc is an ISO Alpha 2 country code";
+  // ISO 3166-1 Alpha 2 country code for the United Kingdom is "GB" but feed names may use "UK".
+  private static final String UK = "UK";
+  private static final String GB = "GB";
+  private final Optional<String> optionalCountryCode;
+
+  private CountryCode(String countryCode) {
+    this.optionalCountryCode = Optional.ofNullable(countryCode);
+  }
+
+  /**
+   * Checks that the given string is a valid ISO Alpha 2 country code (case insensitive).
+   *
+   * <p>We support "UK" as an additional country code equivalent to the ISO 3166-1 Alpha 2 code
+   * "GB".
+   *
+   * @param s the given string
+   * @return true if the given string is a valid ISO Alpha 2 country code (case insensitive)
+   */
+  public static boolean isValidISOAlpha2(String s) {
+    if (s.length() != 2) {
+      return false;
+    }
+    s = s.toUpperCase();
+    return s.equals(UK) || ISO_COUNTRIES.contains(s);
+  }
+
+  /**
+   * Parses a string as a country code name.
+   *
+   * @param countryCode country code for a GTFS dataset
+   * @return a valid GtfsCountryCode object
+   * @throws IllegalArgumentException if illegal @param feedName
+   */
+  public static CountryCode parseString(String countryCode) throws IllegalArgumentException {
+    if (countryCode == null) {
+      return new CountryCode(null);
+    }
+    if (isValidISOAlpha2(countryCode)) {
+      return new CountryCode(countryCode);
+    }
+    throw new IllegalArgumentException(INVALID_COUNTRY_CODE_MESSAGE);
+  }
+
+  public Optional<String> getCountryCode() {
+    return optionalCountryCode;
+  }
+
+  /**
+   * Returns the uppercase ISO 3166-1 Alpha 2 country code component, e.g., "NL" or "AU".
+   *
+   * <p>Note that this returns "GB" for feeds that use the non-standard "UK" country code.
+   *
+   * @return uppercase ISO Alpha country code
+   */
+  public String getISOAlpha2CountryCode() {
+    String countryCode = optionalCountryCode.get().toUpperCase();
```

Comment:
If `optionalCountryCode` is empty `optionalCountryCode.get()` will throw an exception:
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Optional.html#get()

You need to protect `.get()` by wrapping it in a call to `isPresent()` first.

I think a better approach is to move this logic into the constructor and store the uppercase ISO code in `optionalCountryCode` - the only reason it was in a separate method previously was because the code needed to be parsed from the name.

Then, you can get rid of this method so the only way to access the data would be `getCountryCode()`. This would then be a clear indication to the calling code that this value may not be populated (it's `Optional`).